test/e2e/framework/pod/create.go

```diff
@@ -247,13 +250,109 @@ func MakeSecPod(podConfig *Config) (*v1.Pod, error) {
 		volumeIndex++
 	}
 
-	podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
-	podSpec.Spec.Containers[0].VolumeDevices = volumeDevices
-	podSpec.Spec.Volumes = volumes
+	podSpec.Containers[0].VolumeMounts = volumeMounts
+	podSpec.Containers[0].VolumeDevices = volumeDevices
+	podSpec.Volumes = volumes
 	if runtime.GOOS != "windows" {
-		podSpec.Spec.SecurityContext.SELinuxOptions = podConfig.SeLinuxLabel
+		podSpec.SecurityContext.SELinuxOptions = podConfig.SeLinuxLabel
 	}
 
-	SetNodeSelection(&podSpec.Spec, podConfig.NodeSelection)
-	return podSpec, nil
+	SetNodeSelection(podSpec, podConfig.NodeSelection)
+	return podSpec
+}
+
+// NodeOSDistroIs returns true if the distro is the same as `--node-os-distro`
+// the package framework/pod can't import the framework package (see #81245)
+// we need to check if the --node-os-distro=windows is set and the framework package
+// is the one that's parsing the flags, as a workaround this method is looking for the same flag again
+// TODO: replace with `framework.NodeOSDistroIs` when #81245 is complete
+func NodeOSDistroIs(distro string) bool {
```

Comment:
maybe put some of these functions into a util file under pod dir?

Reply:
thanks, I've moved this to a file utils.go in the pod dir